otter/cloud_client.py

```diff
@@ -192,11 +200,85 @@ def got_auth((token, catalog)):
             data=service_request.data,
             params=service_request.params,
             log=log)
-    return auth_eff.on(got_auth)
+
+    eff = auth_eff.on(got_auth)
+    bracket = throttler(service_request.service_type,
+                        service_request.method.lower())
+    if bracket is not None:
+        return Effect(_Throttle(bracket=bracket, effect=eff))
+    else:
+        return eff
+
+
+@attributes(['bracket', 'effect'])
+class _Throttle(object):
+    """
+    A grody hack to allow using a Deferred concurrency limiter in Effectful
+    code.
+
+    A "Deferred bracket" is some function of type
+    ``((f, *args, **kwargs) -> Deferred) -> Deferred``
+    basically, something that "brackets" a call to some Deferred-returning
+    function. This is the case for the ``run`` method of objects like
+    :obj:`.DeferredSemaphore` and :obj:`.DeferredLock`.
+
+    https://wiki.haskell.org/Bracket_pattern
+
+    Ideally Effect would just have built-in concurrency limiters/idioms without
```

Comment:
Why not do this now? Is it too complicated? It's really unfortunate to see deferred mixed with effect in an intent object. How about `effect -> effect` function that does bracketing/throttling? It's performer could use deferred semaphore. Sorry I am sure you might have thought of this. Maybe it is complicated. I'll try to think some more and let you know if something comes up.